gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -354,35 +313,24 @@
           this.existingTargetTable = Optional.absent();
         }
 
-        // Constructing CommitStep object for table registration
-        Path targetPath = getTargetLocation(this.dataset.fs, this.targetFs, this.dataset.table.getDataLocation(),
-            Optional.<Partition> absent());
+        Path targetPath = getTargetLocation(this.targetFs, this.dataset.table.getDataLocation(), Optional.<Partition>absent());
+        if (Boolean.parseBoolean(this.dataset.getProperties().getProperty(ConfigurationKeys.USE_DATASET_LOCAL_WORK_DIR, "false"))) {
```

Comment:
We have a bunch of utils to work with config/properties/maps - ConfigUtils.getBoolean() , State.getPropAsBoolean()
If none of them work here, you can use  MapUtils.getBooleanValue from Apache Commons.